modules/skybrowser/src/wwtcommunicator.cpp

```diff
@@ -0,0 +1,353 @@
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/skybrowser/include/wwtcommunicator.h>
+
+#include <modules/webbrowser/webbrowsermodule.h>
+#include <modules/webbrowser/include/webkeyboardhandler.h>
+#include <modules/webbrowser/include/browserinstance.h>
+#include <ghoul/misc/dictionaryjsonformatter.h> // formatJson
+#include <modules/skybrowser/include/utility.h>
+#include <glm/gtx/vector_angle.hpp>
+#include <cmath> // For atan2
+#define _USE_MATH_DEFINES
+#include <math.h> // For M_PI
+
+namespace {
+    constexpr const char* _loggerCat = "WwtCommunicator";
+
+
+} // namespace
+
+namespace openspace {
+
+    WwtCommunicator::WwtCommunicator(const ghoul::Dictionary& dictionary)
+        : Browser(dictionary) {}
+
+    WwtCommunicator::~WwtCommunicator() {
+
+    }
+
+    void WwtCommunicator::displayImage(const std::string& url, int i)
+    {   
+        // Ensure there are no duplicates
+        auto it = std::find(std::begin(_selectedImages), std::end(_selectedImages), i);
+        if (it == std::end(_selectedImages)) {
+            // Push newly selected image to front
+            _selectedImages.push_front(i);
+            // Index of image is used as layer ID as it is unique in the image data set
+            sendMessageToWwt(addImage(std::to_string(i), url));
+            sendMessageToWwt(setImageOpacity(std::to_string(i), 1.0));
+        }
+    }
+
+    void WwtCommunicator::removeSelectedImage(int i) {
+        // Remove from selected list
+        auto it = std::find(std::begin(_selectedImages), std::end(_selectedImages), i);
+
+        if (it != std::end(_selectedImages)) {
+            _selectedImages.erase(it);
+            sendMessageToWwt(removeImage(std::to_string(i)));
+        }
+    }
+
+    void WwtCommunicator::sendMessageToWwt(const ghoul::Dictionary& msg) {
+        std::string script = "sendMessageToWWT(" + ghoul::formatJson(msg) + ");";
+        executeJavascript(script);
+    }
+
+    const std::deque<int>& WwtCommunicator::getSelectedImages() {
+        return _selectedImages;
+    }
+
+    void WwtCommunicator::setVerticalFov(float vfov) {
+        _verticalFov = vfov;
+        _equatorialAimIsDirty = true;
+    }
+
+    void WwtCommunicator::setWebpageBorderColor(glm::ivec3 color) {
+        std::string stringColor = std::to_string(color.x) + ","
+            + std::to_string(color.y) + "," + std::to_string(color.z);
+        std::string script = "document.body.style.backgroundColor = 'rgb("
+            + stringColor + ")';";
+        executeJavascript(script);
+    }
+
+    void WwtCommunicator::setIsSyncedWithWwt(bool isSynced)
+    {
+        _isSyncedWithWwt = isSynced;
+    }
+
+    void WwtCommunicator::setEquatorialAim(const glm::dvec2& equatorial)
```

Comment:
The parameter can be `glm::dvec2` and then `_equatorialAim = std::move(equatorial);`